### PR TITLE
Option 1: Fix loop that goes out of bounds

### PR DESCRIPTION
**Pull request by devarops**
The `i` in the while loop in `mutate` some times gets bigger than the length of `x`. That is a problem in this line:
```
x[[i]] <- mutate_one(x[[i]])
```
since it produces this error:
```
Error in x[[i]] : subscript out of bounds
```

### CODE REVIEW

**Comment by sckott** [comment]
thanks! 

I wonder if instead we might want `mutate()` to fail if it doesn't create a mutation in any of the functions?  That is, if `i` gets bigger than the length of `x`, then we've gone through all the elements of `x`, correct? Then I imagine we'd just want to `stop()` saying that we couldn't create a mutation?

**Comment by devarops** [comment]
> if `i` gets bigger than the length of `x`, then we've gone through all the elements of x, correct?

Yes, but `mutate_one()` randomly tries [using `astr::ast_modify(...,  if_many = "random")`] a different mutation for each element of `x`. In the second round, `mutate_one` will  most likely choose a  different mutation for the first (and each) element of `x`. In `randgeo`, the example that you are using, `mutate()` succeeds in the first or second round (after going through all the elements of `x` once). This is the current implementation of `mutate()`.

In an alternative implementation of `mutate()`, we could use the first mutation for each element of `x` and then use the second mutation for each element of `x`, and so on. In that case, we would want `mutate()` to fail (`stop()`) if it doesn't create a mutant in any of the functions after systematically trying each mutation.

Which method do you prefer?

- Randomly try a mutation for each function (current)
- Systematically try each mutation for each function (alternative)

This _PR_ solves an issue if you choose to _randomly try a mutation for each function_

We would have to change the implementation of `mutate()` if you chose to _systematically try each mutation for each function_

File: R/mutate.R [hunk 1]
```diff
@@ -28,12 +28,18 @@ mutate <- function(x) {
...
   not_done <- TRUE
   i <- 0
+  j <- 0
   while (not_done) {
-    i <- i + 1
+    i <- (i %% x_length) + 1
+    j <- j + 1
     x[[i]] <- mutate_one(x[[i]])
     if (attr(x[[i]], "mutated")) not_done <- FALSE
+    if (j == max_iterations) break
```
Comment (devarops) [inline R/mutate.R]: @sckott What do you think about this `break`?

**Comment by devarops** [comment]
I have updated this _PR_ adding a `break`.

File: R/mutate.R (hunk 1 above)
Comment (sckott) [reply R/mutate.R]: looks good, i've usually used `while`, but we can try this

**Comment by sckott** [comment]
Let's stick with random for now.

**Comment by devarops** [comment]
Great!

**Pull request merged**